mbedtls/tests/support/keys.rs

```diff
@@ -92,3 +92,7 @@ pub const ROOT_CA_KEY: &'static str = concat!(include_str!("./keys/ca.key"),"\0"
 pub const EXPIRED_CERT_SUBJECT: &'static str = "CN=ExpiredNode";
 pub const EXPIRED_CERT: &'static str = concat!(include_str!("./keys/expired.crt"),"\0");
 pub const EXPIRED_KEY: &'static str = concat!(include_str!("./keys/expired.key"),"\0");
+
+pub const PRESHARED_KEY: &'static [u8] = &[
+    234, 206, 151, 23, 219,  21,  71, 144,
+    107,  42,  23, 67, 249, 173, 182, 224 ];
```

Comment:
Could you pick a preshared key that is not valid utf8 and also contains a zero byte? It may trigger some weird code paths.

Reply:
Also, could we add a case with a leading zero byte? I'm certain it is correctly handled, but I have seen way too many leading zero bugs in the wild/

(Orthogonal but also see https://mbed-tls.readthedocs.io/en/latest/security-advisories/advisories/mbedtls-security-advisory-2020-09-3/).

Reply:
Huh! If it's not valid UTF-8, we should expect an error from `from_utf8`?

What kind of error do we expect for leading zero bytes?

Reply:
Why is there a requirement that pre-shared keys are valid UTF-8? At least in [RFC4279](https://datatracker.ietf.org/doc/html/rfc4279), only PSK _identities_ are assumed to be UTF-8 (Sec. 5.1). Also Sec. 4 states that PSK are the result of Diffie Hellman computations, so no UTF-8 expected.